db/src/schema.rs

```diff
@@ -52,6 +55,135 @@ fn validate_schema_map(entid_map: &EntidMap, schema_map: &SchemaMap) -> Result<(
     Ok(())
 }
 
+#[derive(Clone,Debug,Default,Eq,Hash,Ord,PartialOrd,PartialEq)]
+pub struct AttributeBuilder {
+    value_type: Option<ValueType>,
+    multival: Option<bool>,
+    unique_value: Option<bool>,
+    unique_identity: Option<bool>,
+    index: Option<bool>,
+    fulltext: Option<bool>,
+    component: Option<bool>,
+}
+
+impl AttributeBuilder {
+    pub fn value_type<'a>(&'a mut self, value_type: ValueType) -> &'a mut Self {
```

Comment:
Why mutable references rather than ownership? Surely the model is

```
let foo = AttributeBuilder::new().value_type(ValueType::Boolean).build();
```

which can be modeled as

```
pub fn value_type(self, value_type: ValueType) -> Self {
```

no?